elastic/kibana
# [Discover] Address "Don't call Hooks" React warnings

Closes https://github.com/elastic/kibana/issues/131621

## Summary

This PR addresses React warnings which we noticed thanks to Jest output.

Steps to test:
* Check the jest output: such warnings should not be present 
* Check that the fields doc popover still works

<img width="489" alt="Screenshot 2022-06-14 at 14 50 13" src="https://user-images.githubusercontent.com/1415710/173581159-7c10ed15-2cc3-48a0-aac9-bdd7b569c82d.png">



### Checklist
